lib/vimeo.js

```diff
@@ -570,14 +570,23 @@ Vimeo.prototype.upload = function (
  *
  * https://developer.vimeo.com/api/reference/videos#create_video_version
  *
- * @param {string}    file              Path to the file you wish to upload.
- * @param {string}    videoUri          Video URI of the video file to replace.
- * @param {Object=}   params            Parameters to send when creating a new video (name,
- *                                      privacy restrictions, etc.). See the API documentation for
- *                                      supported parameters.
- * @param {Function}  completeCallback  Callback to be executed when the upload completes.
- * @param {Function}  progressCallback  Callback to be executed when upload progress is updated.
- * @param {Function}  errorCallback     Callback to be executed when the upload returns an error.
+ * .replace( file, videoUri [, params] [, completeCallback], progressCallback [, errorCallback])
+ *
+ * -  params (optional)
+ *    If an object is not provided, default upload params are used.
+ *
+ * -  completeCallback and errorCallback (optional)
+ *    If neither passed in, a Promise will be returned.
+ *    Ex. vimeo.replace(file, videoUri, progressCallback) or vimeo.replace(file, videoUri, params, progressCallback)
+ *
+ * @param {string}    file                Path to the file you wish to upload.
+ * @param {string}    videoUri            Video URI of the video file to replace.
+ * @param {Object=}   [params]            (optional) Parameters to send when creating a new video (name,
+ *                                        privacy restrictions, etc.). See the API documentation for
+ *                                        supported parameters.
+ * @param {Function}  [completeCallback]  (optional) Callback to be executed when the upload completes.
+ * @param {Function}  progressCallback    Callback to be executed when upload progress is updated.
+ * @param {Function}  [errorCallback]     (optional) Callback to be executed when the upload returns an error.
```

Comment:
I don't think we can change this...but the function signature on this is so awkward, with required parameters after optional ones.

Reply:
haha yea..I think `completeCallback` was given a higher priority so it was the first.